crates/ty_python_semantic/src/semantic_index/builder.rs

```diff
@@ -2654,3 +2654,76 @@ fn dunder_all_extend_argument(value: &ast::Expr) -> Option<&ast::Expr> {
 
     (attr == "__all__").then_some(value)
 }
+
+/// Builds an interval-map that matches expressions (by their node index) to their enclosing scopes.
+///
+/// The interval map is built in a two-step process because the expression ids are assigned in source order,
+/// but we visit the expressions in semantic order. Few expressions are registered out of order.
```

Comment:
Is this something that would change with the proposed change in https://github.com/astral-sh/ruff/pull/19271?

Reply:
Yes, but I don't think it would invalidate the entire approach. Instead, we would have to use a regular `sort` call in `build` before building the interval map (and Rust's sorting claims to be pretty good at sorting mostly sorted data)